src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/AppointmentService.kt

```diff
@@ -25,57 +31,100 @@ class AppointmentService(
     durationInMinutes: Int,
     appointmentTime: OffsetDateTime,
     appointmentType: AppointmentType,
-    createdByUser: AuthUser
+    createdByUser: AuthUser,
+    appointmentDeliveryType: AppointmentDeliveryType,
+    appointmentDeliveryAddress: AddressDTO? = null,
   ): Appointment {
 
     val initialAppointmentRequired = appointment == null
     if (initialAppointmentRequired) {
       val deliusAppointmentId = communityAPIBookingService.book(referral, null, appointmentTime, durationInMinutes, appointmentType)
-      return createAppointment(durationInMinutes, appointmentTime, deliusAppointmentId, createdByUser)
+      return createAppointment(durationInMinutes, appointmentTime, deliusAppointmentId, createdByUser, appointmentDeliveryType, appointmentDeliveryAddress)
     }
 
     val updateCurrentAppointmentRequired = appointment!!.attended == null
     if (updateCurrentAppointmentRequired) {
```

Comment:
_this does not block this PR, can be discussed after it's merged_

Can we remove all branching here, by creating a new appointment every time `createOrUpdateAppointment` is called?

Ie. instead of figuring out create-or-update, we can always create an appointment and make the previous one superseded? It would significantly reduce the complexity here

Reply:
yeah, I think this part of the code can be improved.
(Although this code was not part of changes I made)
I'll add a TODO here